salloc, sattach, sbcast, sdiag, sprio, sshare, strigger, sview, sh5util, scrun, scrontab: add pages

gutjuri opened the pull request:
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):** 23.11

kbdharun commented on pages/linux/sh5util.md:
```diff
@@ -0,0 +1,16 @@
+> Merge HDF5 files (produced by the `sacct_gather_profile` plugin).
+> More information: <https://slurm.schedmd.com/sh5util.html>.
+
+- Merge HDF5 files produced on each allocated node for the specified job (-step):
```
Kinda curious, what does `(-step)` mean here, does it mean job step?

kbdharun commented on pages/linux/sh5util.md:
```diff
@@ -0,0 +1,16 @@
+# sh5util
+
+> Merge HDF5 files (produced by the `sacct_gather_profile` plugin).
```
Is it necessary that this part should be inside brackets?

```suggestion
> Merge HDF5 files produced by the `sacct_gather_profile` plugin.
```

kbdharun commented on pages/linux/scrontab.md:
```diff
@@ -0,0 +1,20 @@
+> Manage Slurm crontab files.
+> More information: <https://slurm.schedmd.com/scrontab.html>.
+
+- Install a new crontab from the specified file:
```
Should we place crontab inside backticks here or is it fine as it is?

gutjuri replied:
It is not used in the sense of the command `crontab`, but in the sense of the concept of a crontab. That's why i didn't put it in backticks.

kbdharun reviewed (approved):
LGTM, Thanks for your contribution. I have a few queries after which this PR is GTG.

kbdharun commented:
Before merging this PR, I would suggest waiting for @sbrl's review to checkout the commands, as they use Slurm in their university.

sbrl commented:
Wow, this is an enormous PR! I'll check this thoroughly tomorrow, as it's 3:30am here right now.

sbrl commented on pages/linux/salloc.md:
```diff
@@ -0,0 +1,16 @@
+# salloc
+
+> Allocate nodes in a cluster and start an interactive shell session or execute a command.
```
This doesn't really allocate nodes though? It allocates a job on a node.
```suggestion
> Start an interactive shell session or execute a command by allocating on 1 or more nodes in a cluster. 
```

sbrl commented on pages/linux/sattach.md:
```diff
@@ -0,0 +1,16 @@
+> Attach to a Slurm job step.
+> More information: <https://slurm.schedmd.com/sattach.html>.
+
+- Make available the IO streams (`stdout`, `stderr`, and `stdin`) of a Slurm job step:
```
Define `Make available` here.

gutjuri replied:
```suggestion
-  Redirect the IO streams (`stdout`, `stderr`, and `stdin`) of a Slurm job step to the current terminal:
```

Do you think this would be better?

sbrl replied:
Yes, that looks much better @gutjuri. Many thanks!

sbrl commented on pages/linux/sattach.md:
```diff
@@ -0,0 +1,16 @@
+
+- Transmit `stdin` to the specified task only:
+
+`sattach --input-filter {{task_number}}`
```
What do you mean by `task number`? Is this a job number, or something else?

Try to use the official language from the project itself.

Further, what do you mean by transmit to the specified task? I'm rather confused. Is this attaching interactively, or pointing something else at a job's stdin?

gutjuri replied:
"Task" is an official term. I cannot find a document providing an exact definition, but the term is mentioned several times in the docs, e.g. here https://slurm.schedmd.com/cpu_management.html#Section2

See also: https://stackoverflow.com/questions/39186698/what-does-the-ntasks-or-n-tasks-does-in-slurm

sbrl replied:
Hmm okay, that makes sense. Thanks for the clarification.

Could you explain what you mean by `transmit to the specified task` though please?

gutjuri replied:
Basically that I send input from the current console to the specified task.

sbrl commented on pages/linux/sattach.md:
```diff
@@ -0,0 +1,16 @@
+
+`sattach --input-filter {{task_number}}`
+
+- Only display `stdin`/`stderr` of the specified task:
```
Please try to keep the language consistent. In this page alone you've used `display`, `transmit`, and `make available`. This is very confusing.

gutjuri replied:
```suggestion
- Only redirect `stdin`/`stderr` of the specified task:
```

sbrl commented on pages/linux/sbcast.md:
```diff
@@ -0,0 +1,13 @@
+
+- Autodetect shared libraries the transmitted file depends upon and transmit them as well:
+
+`sbcast --send-libs={{yes}} {{path/to/executable}} {{path/to/destination}}`
```
Do you have a link to documentation on this? All the documentation I've seen for `sbcast` doesn't have `--send-libs` as an argument.

gutjuri replied:
https://slurm.schedmd.com/sbcast.html#SECTION_OPTIONS

sbrl commented on pages/linux/scrontab.md:
```diff
@@ -0,0 +1,20 @@
+
+`scrontab {{path/to/file}}`
+
+- [e]dit the crontab of the specified user:
+
+`scrontab --user={{user_id}} -e`
```
Should we also have an example here for interactively editing the crontab of the current user?

gutjuri replied:
You're right, i'll add one.

sbrl commented on pages/linux/scrontab.md:
```diff
@@ -0,0 +1,20 @@
+
+`scrontab -r`
+
+- [l]ist the current crontab to `stdout`:
```
Need to be specific here given previous examples.
```suggestion
- Print the crontab of the current user to `stdout`:
```

sbrl commented on pages/linux/scrun.md:
```diff
@@ -0,0 +1,24 @@
+
+`scrun create {{container_id}}`
+
+- Start a container in a Slurm job:
+
+`scrun start {{container_id}}`
```
Is there a way to pass arguments to a container at all? It might be worth having an extra example directly beneath this one about that if so.

sbrl commented on pages/linux/scrun.md:
```diff
@@ -0,0 +1,24 @@
+> An OCI (Open Container Initiative) runtime proxy for Slurm.
+> More information: <https://slurm.schedmd.com/scrun.html>.
+
+- Create a container in the current directory:
+
+`scrun create {{container_id}}`
```
What do yo  mean by create a container here? Are we creating it from nothing? Where does the image come from? Is this like `docker build`, or does it download the container from somewhere?

It would be nice if this example was specific about where the container comes from.

gutjuri replied:
Looks like I misunderstood `scrun`. It seems like scrun is not meant as an interface to users (https://slurm.schedmd.com/SLUG23/containers-slug23.pdf, slide 14), but for the developers of e.g. docker, podman etc (?). 

For me, this raises the question on whether to include `scrun` in tldr at all.

sbrl replied:
Hmmm, interesting.

I don't see why it shouldn't be included, even if it isn't user-facing @gutjuri. But I would suggest it isn't a priority on including it vs eg. `sbatch` for example.

Your call.

gutjuri replied:
I'll remove the page then and hope someone with experience using this command will create it.

sbrl commented on pages/linux/sdiag.md:
```diff
@@ -0,0 +1,20 @@
+> Show information about the execution of `slurmctld`.
+> More information: <https://slurm.schedmd.com/sdiag.html>.
+
+- Show several performance counters related to the execution of `slurmctld`:
```
Again with the accuracy and specificity of language here.
```suggestion
- Show all performance counters related to the execution of `slurmctld`:
```

sbrl commented on pages/linux/sprio.md:
```diff
@@ -0,0 +1,20 @@
+> View the factors determining a job's scheduling priority.
+> More information: <https://slurm.schedmd.com/sprio.html>.
+
+- View the factors determining all pending job's scheduling priority:
```
Grammar
```suggestion
- View the factors determining the scheduling priority of all jobs:
```

sbrl commented on pages/linux/sprio.md:
```diff
@@ -0,0 +1,20 @@
+
+`sprio --jobs={{job_id_1,job_id_2,...}}`
+
+- View information for the jobs of specified users and output additional information:
+
+`sprio --user={{user_name_1,user_name_2,...}} --long`
```
`--user` and `--long` should be 2 separate examples.

sbrl commented on pages/linux/strigger.md:
```diff
@@ -0,0 +1,24 @@
+# strigger
+
+> View or modify Slurm trigger information.
```
An extract description line of context would be useful here:
```suggestion
> View or modify Slurm trigger information.
> Triggers are actions that are automatically run when a given event occurs on a Slurm cluster.
```

sbrl commented on pages/linux/strigger.md:
```diff
@@ -0,0 +1,24 @@
+
+- Execute the specified program when the specified job terminated:
+
+`strigger --set --jobid={{job_id}} --fini --program={{path/to/executable}}`
```
Can you pass arguments too? If so, might be worth an extra example here.

gutjuri replied:
The docs state

1. "A hostlist expression for the nodelist or job ID is passed as an argument to the program."
2. "You may quote the path and include extra program arguments if desired."

So I think that you could do

```suggestion
`strigger --set --jobid={{job_id}} --fini --program="{{path/to/executable}} {{argument1 argument2 ...}}"`
```

Do you think that that's correct?

sbrl replied:
That seems good. Let's do that, @gutjuri.

sbrl reviewed (changes requested):
Wow, thanks for the new pages @gutjuri! I've left some comments below for you to review.

Sorry there are a lot of comments here - you're contributing a lot of pages at once, so nautrally there will be more review comments :-)

gutjuri commented on pages/linux/sattach.md:
```diff
@@ -0,0 +1,16 @@
+
+`sattach {{jobid}}.{{stepid}}`
+
+- Transmit `stdin` to the specified task only:
```
```suggestion
- Use the current console's input as `stdin` to the specified task:
```

Not sure how to formulate this better.

sbrl replied:
That suggestion reads much better - let's do that.

gutjuri commented on pages/linux/scrontab.md:
```diff
@@ -0,0 +1,20 @@
+
+`scrontab {{path/to/file}}`
+
+- [e]dit the crontab of the specified user:
+
+`scrontab --user={{user_id}} -e`
```
```suggestion
- [e]dit the crontab of the current user:

`scrontab -e`

- [e]dit the crontab of the specified user:

`scrontab --user={{user_id}} -e`
```

gutjuri commented on pages/linux/sprio.md:
```diff
@@ -0,0 +1,20 @@
+
+`sprio --jobs={{job_id_1,job_id_2,...}}`
+
+- View information for the jobs of specified users and output additional information:
+
+`sprio --user={{user_name_1,user_name_2,...}} --long`
```
```suggestion
- Output additional information:

`sprio --long`

- View information for the jobs of specified users:

`sprio --user={{user_name_1,user_name_2,...}}`
```

gutjuri commented:
> Wow, thanks for the new pages @gutjuri! I've left some comments below for you to review.
> 
> Sorry there are a lot of comments here - you're contributing a lot of pages at once, so nautrally there will be more review comments :-)

Thanks for your detailed feedback. I think I have addressed all of the issues you raised in my comments. Could you check out my suggestions and comment or add a thumbs up if you think they're fine?

sbrl commented:
Looks good, thanks for the hard work @gutjuri! Got a few more comments there for you. I think we're quite close!

sbrl reviewed (approved):
Looks good to me! Thanks so much for your hard work on this, @gutjuri! Large PRs like this was that add lots of pages can be a challenge - thanks for persevering with the long review process!
